### ISSUE TITLE
Fix broken tests because of gasPrice (0)

### ISSUE THREAD

**Issue by udosson** [kumodao/borrowprot#12]
The smart contract tests currently fail with the error message `Error: Returned error: Transaction gasPrice (0) is too low for the next block, which has a baseFeePerGas of 8`

There is a similar issue at liquity but the error hasn't been fixed yet.
https://github.com/liquity/dev/issues/765

- [x] BorrowerOperations
- [x] CollSurplusPool
- [x] Gas compensation tests
- [x] LQTYStaking revenue share tests
- [x] BorrowerWrappers
- [x] StabilityPool - LQTY Rewards
- [x] StabilityPool
- [x] TroveManager - in Recovery Mode
- [x] TroveManager

**Comment by udosson** [comment]
I fixed one test in this commit https://github.com/udosson/borrowprot/commit/e4dce999c6a001665ad87908ddfd1b9625906e1e
The pattern should be the same for all failing tests.
When Liquity developed their project, it was still possible to send transactions without gas in the local hardhat network. This changed however with EIP-1559.
Two things we need to do:
1. provide sufficient gas for transactions (e.g. https://github.com/udosson/borrowprot/blob/fix-broken-tests-gas/packages/contracts/test/StabilityPoolTest.js#L1000)
2. adjust the amount of ETH of the corresponding account (e.g. https://github.com/udosson/borrowprot/blob/fix-broken-tests-gas/packages/contracts/test/StabilityPoolTest.js#L1007)

**Comment by wollooo** [comment]
I did no further work after our joint session on Wednesday, but will look at TroveManager over the Weekend. 
The tests, which are still faulty, can be seen in the following log: `https://github.com/wollooo/borrowprot/runs/5128638437?check_suite_focus=true
`
What puzzles me a bit is that many tests of LQTYStakingRewardTest fail even though we have fixed it together. However, when I test the file individually, everything runs. (See this log: https://github.com/wollooo/borrowprot/runs/5156736595?check_suite_focus=true)

Maybe you can take a close look at the other bugs. In the first log these are errors 1-17 and concern the first 3 not fixed files in the first post of this issue.

**Comment by udosson** [comment]
@wollooo thanks for your investigation. 
The error occurs when calling the following function (just as an example, the errors follow the same pattern)
`const redemptionTx = await th.redeemCollateralAndGetTxObject(B, contracts, dec(100, 18))` https://github.com/wollooo/borrowprot/blob/main/packages/contracts/test/LQTYStakingFeeRewardsTest.js#L111

which calls
`static async redeemCollateralAndGetTxObject(redeemer, contracts, LUSDAmount, maxFee = this._100pct) ` 
https://github.com/wollooo/borrowprot/blob/3e8174c4870502ce09f7bee6270a1de0cced4982/packages/contracts/utils/testHelpers.js#L950

which calls
`static async performRedemptionTx(redeemer, price, contracts, LUSDAmount, maxFee = 0)`
https://github.com/wollooo/borrowprot/blob/3e8174c4870502ce09f7bee6270a1de0cced4982/packages/contracts/utils/testHelpers.js#L970

and in this function, the following call causes the error:
```
const tx = await contracts.troveManager.redeemCollateral(LUSDAmount,
      firstRedemptionHint,
      exactPartialRedemptionHint[0],
      exactPartialRedemptionHint[1],
      partialRedemptionNewICR,
      0, maxFee,
      { from: redeemer, gasPrice: 0 },
    )
```
https://github.com/wollooo/borrowprot/blob/3e8174c4870502ce09f7bee6270a1de0cced4982/packages/contracts/utils/testHelpers.js#L992

We could either pass the gas price from `LQTYStakingFeeRewardsTest.js` or set it in `./utils/testHelpers.js`.
I prefer the first option because then all tests follow the same pattern.

**Comment by udosson** [comment]
@wollooo thank you so much for fixing all the broken tests.
I fixed the missing ones in this commit https://github.com/udosson/borrowprot/commit/4cd2f1c0738713eae03f31bed371132ea4d8d60e

I subtracted the amount of gas used from the `expectedReceivedETH`.
const expectedReceivedETH = expectedTotalETHDrawn.sub(toBN(ETHFee)).sub(toBN(th.gasUsed(redemptionTx) * GAS_PRICE))

Feel free to use my commit and create a PR. I'm looking forward to merging it.
